client/chat/channel.tsx

```diff
@@ -171,151 +167,121 @@ const UserListEntry = React.memo<UserListEntryProps>(props => {
-            Offline ({offline.size})
-          </UserListOverline>
-        )
-      } else if (i < offline.size + 1) {
-        const username = offline.get(i - 1)!
-        return <UserListEntry style={style} username={username} key={username} faded={true} />
+  const renderRow = useCallback(
+    ({ index, style }: { index: number; style: React.CSSProperties }) => {
+      if (index > rowCount) {
+        throw new Error('Asked to render nonexistent user: ' + index)
       }
 
-      i -= offline.size + 1
-    }
-
-    if (i === 0) {
-      return <UserListPadding style={style} key={index} />
-    }
-
-    throw new Error('Asked to render nonexistent user: ' + index)
-  }
+      const idleHeaderIndex = idle.size ? active.size + 1 : null
+      let offlineHeaderIndex = null
+      if (offline.size) {
+        offlineHeaderIndex =
+          idleHeaderIndex != null ? idleHeaderIndex + idle.size + 1 : active.size + 1
+      }
 
-  render() {
-    const { active, idle, offline } = this.props.users
-    const rowCount =
-      1 + active.size + (idle.size ? 1 : 0) + idle.size + (offline.size ? 1 : 0) + offline.size + 1
+      switch (index) {
+        case 0:
+          return (
+            <UserListOverline style={style} key={index}>
+              Active ({active.size})
+            </UserListOverline>
+          )
+        case idleHeaderIndex:
+          return (
+            <UserListOverline style={style} key={index}>
+              Idle ({idle.size})
+            </UserListOverline>
+          )
+        case offlineHeaderIndex:
+          return (
+            <UserListOverline style={style} key={index}>
+              Offline ({offline.size})
+            </UserListOverline>
+          )
+        default:
+          let username: string | undefined
+          let faded = false
+          if (index < active.size + 1) {
+            username = active.get(index - 1)!
+          } else if (offlineHeaderIndex && index > offlineHeaderIndex) {
+            faded = true
+            username = offline.get(index - offlineHeaderIndex - 1)!
+          } else {
+            username = idleHeaderIndex ? idle.get(index - idleHeaderIndex - 1)! : undefined
+          }
+
+          if (username) {
+            return <UserListEntry style={style} username={username} key={username} faded={faded} />
+          }
+          throw new Error('Asked to render nonexistent user: ' + index)
+      }
+    },
+    [active, idle, offline, rowCount],
+  )
 
-    return (
-      <UserListContainer ref={this._contentRef}>
-        <WindowListener event='resize' listener={this.updateDimensions} />
-        <UsersVirtualizedList
-          ref={this._listRef}
-          width={this.state.width}
-          height={this.state.height}
-          rowCount={rowCount}
-          rowHeight={this.getRowHeight}
-          rowRenderer={this.renderRow}
-        />
-      </UserListContainer>
-    )
-  }
+  listRef.current?.resetAfterIndex(0, false)
```

Comment:
Might be good to memoize this (I think a good candidate for the `deps` to use would be `getRowHeight`, since the values for row height can only change if that's changed)

Reply:
The linter warns that `getRowHeight` is an unnecessary dependency and to remove it

Reply:
You can add a `// eslint-disable-next-line` comment to disable that particular warning (VSCode should have an action to do this on the offending line)